src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,935 @@
+
+        JsonNode json = doPost(SESSION, request, JsonNode.class);
+
+        if (getJsonProperty(json, "success").asBoolean()) {
+            sessionId = getJsonProperty(json, "sessionId").asText();
+        } else {
+            throw new FailedLoginException("Unable to login.");
+        }
+    }
+
+    @Override
+    protected HttpHeaders putExtraRequestHeaders(HttpMethod httpMethod, String uri, HttpHeaders headers) throws Exception {
+        if (sessionId != null && !uri.equals(SESSION)) {
+            headers.add("Cookie", String.format("session_id=%s; Path=/; Domain=%s; Secure; HttpOnly;", sessionId, getHost()));
+        }
+        return super.putExtraRequestHeaders(httpMethod, uri, headers);
+    }
+
+    @Override
+    public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+        String property = controllableProperty.getProperty();
+        String value = String.valueOf(controllableProperty.getValue());
+
+        controlOperationsLock.lock();
+        try {
+            switch (property) {
+                case "Mute Microphones":
+                    if (value.equals("0")) {
+                        unmute();
+                    } else {
+                        mute();
+                    }
+                    updateLatestControlTimestamp();
+                    break;
+                case "Mute Local Video":
+                    setMuteVideoStatus(value.equals("1"));
+                    break;
+                case "Audio Volume":
+                    setVolume(Math.round(Float.parseFloat(value)));
+                    updateLatestControlTimestamp();
+                    break;
+                case "Reboot":
+                    reboot();
+                    break;
+                default:
+                    break;
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public void controlProperties(List<ControllableProperty> list) throws Exception {
+        if (CollectionUtils.isEmpty(list)) {
+            throw new IllegalArgumentException("Controllable properties cannot be null or empty");
+        }
+
+        for (ControllableProperty controllableProperty : list) {
+            controlProperty(controllableProperty);
+        }
+    }
+
+    /**
+     * Interceptor for RestTemplate that injects
+     * Cookie header and reacts to the communicator state - after reboot the certificates on the device get
+     * refreshed that results with timeouts on https requests. In order to overcome this - communicator is being internally
+     * destroyed on the successful reboot action. Hovewer, we cannot guarantee that any other resource does not issue
+     * reboot or certificates do not get refreshed for other reasons, so we check for a specific error on
+     * authorization event and reinitialize the communicator here.
+     */
+    class PolycomVideoOSInterceptor implements ClientHttpRequestInterceptor {

```

Comment:
Lets review this class when we talk...I want to make sure I understand specifically the error handling.